src/Parser/Shaman/Elemental/Modules/ShamanCore/EarthShock.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import ItemLink from 'common/ItemLink';
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { formatNumber } from 'common/format';
+
+class EarthShock extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+    combatants: Combatants,
+  };
+
+  maelstromSpendOutAscendence = [];
+  maelstromSpendInAscendence = [];
+  latershock = false;
+  gambling = false;
+
+  on_initialized() {
+    this.gambling = (this.combatants.selected.hasHands(ITEMS.SMOLDERING_HEART.id) && this.combatants.selected.hasFeet(ITEMS.THE_DECEIVERS_BLOOD_PACT.id));
+    this.pseudogambling = (this.combatants.selected.hasShoulder(ITEMS.ECHOES_OF_THE_GREAT_SUNDERING.id) && this.combatants.selected.hasFeet(ITEMS.THE_DECEIVERS_BLOOD_PACT.id));
+  }
+
+  get avgMaelstromSpendOutOfAsc() {
+    let spend = this.maelstromSpendOutAscendence.reduce((prev, cur) => {
+      return prev + cur;
+    }, 0);
+    spend /= this.maelstromSpendOutAscendence.length || 1;
+    return spend;
+  }
+  get avgMaelstromSpendInAsc() {
+    let spend = this.maelstromSpendInAscendence.reduce((prev, cur) => {
+      return prev + cur;
+    },0);
+    spend /= this.maelstromSpendInAscendence.length||1;
+    return spend;
+  }
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid === SPELLS.EARTH_SHOCK.id && !this.latershock) {
+      this.latershock = true;
+      if (event.classResources[0].type === RESOURCE_TYPES.MAELSTROM.id) {
+        if (this.combatants.selected.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id)) {
```

Comment:
I assume you want to check if the player has Ascendance at the moment of cast. If yes, you need to supply additional parameter to `hasBuff` like this: 
```jsx
if (this.combatants.selected.hasBuff(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id, event.timestamp))
```